src/github/dto/update.practices.dto.ts

```diff
@@ -0,0 +1,11 @@
+import { IsNotEmpty } from 'class-validator';
+import { ApiProperty } from '@nestjs/swagger';
+
+export class UpdatePracticesDto {
+  @ApiProperty({
+    description: 'Practices can be strings or a single paragraph.',
+    example: ['practice1', 'practice2'],
+  })
+  @IsNotEmpty()
+  practices: string[];
```

Comment:
PullProofAI

Using `@IsNotEmpty()` without `{ each: true }` on an array field like `practices` will only check if the array itself is not empty, not validating each individual element. This could lead to unexpected behavior if the array contains empty strings or null values. It's recommended to use `@IsArray()` along with `@IsNotEmpty({ each: true })` to ensure each element in the array is validated properly.

```diff
-   @IsNotEmpty()
+   @IsArray()
+   @IsNotEmpty({ each: true })
```


<!-- This is an auto-generated comment by PullProofAI -->